File: ticket-backend/src/main/java/aics/domain/fuzzy/models/FuzzyVariableRating.java
package aics.domain.fuzzy.models;

import aics.domain.fuzzy.constants.FuzzyVariableRatingFields;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.experimental.Accessors;

import java.io.Serializable;
import java.util.HashMap;

@Data
@Accessors(chain = true)
@AllArgsConstructor
public class FuzzyVariableRating implements Serializable, FuzzyVariableI {
    private FuzzyVariableDistributionPart varBad;
    private FuzzyVariableDistributionPart varAverage;
    private FuzzyVariableDistributionPart varGood;
    private FuzzyVariableDistributionPart varVeryGood;

    @Override
    public HashMap<String, FuzzyVariableDistributionPart> getFuzzyVariableMap() {
        HashMap<String, FuzzyVariableDistributionPart> fuzzyVariableMap = new HashMap<>();
        fuzzyVariableMap.put(FuzzyVariableRatingFields.BAD.name(), this.varBad);
        fuzzyVariableMap.put(FuzzyVariableRatingFields.AVERAGE.name(), this.varAverage);
        fuzzyVariableMap.put(FuzzyVariableRatingFields.GOOD.name(), this.varGood);
        fuzzyVariableMap.put(FuzzyVariableRatingFields.VERY_GOOD.name(), this.varVeryGood);
        return fuzzyVariableMap;
    }
}

File: ticket-frontend/src/modules/fuzzy/fuzzy-variable-rating.ts
import { FuzzyVariableDistributionPart, FuzzyVariableDistributionPartUtils, FuzzyVariableI } from './fuzzy-variable-distribution';

export class FuzzyVariableRating implements FuzzyVariableI {
    varBad: FuzzyVariableDistributionPart;
    varAverage: FuzzyVariableDistributionPart;
    varGood: FuzzyVariableDistributionPart;
    varVeryGood: FuzzyVariableDistributionPart;

    constructor(obj: {
        varBad: FuzzyVariableDistributionPart,
        varAverage: FuzzyVariableDistributionPart,
        varGood: FuzzyVariableDistributionPart,
        varVeryGood: FuzzyVariableDistributionPart
    }) {
        this.varBad = obj.varBad;
        this.varAverage = obj.varAverage;
        this.varGood = obj.varGood;
        this.varVeryGood = obj.varVeryGood;
    }

    static fromObj(obj: any): FuzzyVariableRating {
        return new FuzzyVariableRating({
            varBad: FuzzyVariableDistributionPartUtils.fuzzyVariableDistributionPartFromObj(obj.varBad),
            varAverage: FuzzyVariableDistributionPartUtils.fuzzyVariableDistributionPartFromObj(obj.varAverage),
            varGood: FuzzyVariableDistributionPartUtils.fuzzyVariableDistributionPartFromObj(obj.varGood),
            varVeryGood: FuzzyVariableDistributionPartUtils.fuzzyVariableDistributionPartFromObj(obj.varVeryGood)
        })
    }

    getFuzzyVariableMap(): Record<string, FuzzyVariableDistributionPart> {
        return {
            [FuzzyVariableRatingFields.BAD]: this.varBad,
            [FuzzyVariableRatingFields.AVERAGE]: this.varAverage,
            [FuzzyVariableRatingFields.GOOD]: this.varGood,
            [FuzzyVariableRatingFields.VERY_GOOD]: this.varVeryGood,
        }
    }

    getFuzzyVariableColorsMap(): Record<string, string> {
        return {
            [FuzzyVariableRatingFields.BAD]: 'red',
            [FuzzyVariableRatingFields.AVERAGE]: 'orange',
            [FuzzyVariableRatingFields.GOOD]: 'yellow',
            [FuzzyVariableRatingFields.VERY_GOOD]: 'green',
        }
    }
}

export enum FuzzyVariableRatingFields {
    BAD = 'BAD',
    AVERAGE = 'AVERAGE',
    GOOD = 'GOOD',
    VERY_GOOD = 'VERY_GOOD',
}